crates/holochain_sqlite/src/db/p2p_agent_store.rs

```diff
@@ -254,58 +410,17 @@ impl AsP2pStateTxExt for Transaction<'_> {
         until_ms: u64,
         arcset: DhtArcSet,
     ) -> DatabaseResult<Vec<AgentInfoSigned>> {
-        let mut stmt = self
-            .prepare(sql_p2p_agent_store::GOSSIP_QUERY)
-            .map_err(|e| rusqlite::Error::ToSqlConversionFailure(e.into()))?;
-        let mut out = Vec::new();
-        for r in stmt.query_map(
-            named_params! {
-                // TODO: just take i64 so we don't need to clamp (probably
-                //       should just do more Timestamp refactor)
-                ":since_ms": clamp64(since_ms),
-                ":until_ms": clamp64(until_ms),
-                // we filter by arc in memory, not in the db query
-                ":storage_start_loc": Some(u32::MIN),
-                ":storage_end_loc": Some(u32::MAX),
-            },
-            |r| {
-                let r = r.get_ref(0)?;
-                let r = r.as_blob()?;
-                let signed = AgentInfoSigned::decode(r)
-                    .map_err(|e| rusqlite::Error::ToSqlConversionFailure(e.into()))?;
-
-                Ok(signed)
-            },
-        )? {
-            let info = r?;
-            let interval = DhtArcRange::from(info.storage_arc);
-            if arcset.overlap(&interval.into()) {
-                out.push(info);
-            }
-        }
-        Ok(out)
+        ai_cache(self)?.query_agents(since_ms, until_ms, arcset)
     }
 
     fn p2p_query_near_basis(&self, basis: u32, limit: u32) -> DatabaseResult<Vec<AgentInfoSigned>> {
-        let mut stmt = self
-            .prepare(sql_p2p_agent_store::QUERY_NEAR_BASIS)
-            .map_err(|e| rusqlite::Error::ToSqlConversionFailure(e.into()))?;
-
-        let mut out = Vec::new();
-        for r in stmt.query_map(named_params! { ":basis": basis, ":limit": limit }, |r| {
-            let r = r.get_ref(0)?;
-            let r = r.as_blob()?;
-            let signed = AgentInfoSigned::decode(r)
-                .map_err(|e| rusqlite::Error::ToSqlConversionFailure(e.into()))?;
-
-            Ok(signed)
-        })? {
-            out.push(r?);
-        }
-        Ok(out)
+        ai_cache(self)?.query_near_basis(basis, limit)
     }
 
     fn p2p_extrapolated_coverage(&self, dht_arc_set: DhtArcSet) -> DatabaseResult<Vec<f64>> {
+        // TODO - rewrite this to use the ai_cache memcached info
```

Comment:
Maybe good to avoid memcached because that's a product name? Possibly misleading

Reply:
```suggestion
        // TODO - rewrite this to use the "ai_cache" memory cached info
```